node/src/components/chainspec_loader.rs

```diff
@@ -258,7 +258,7 @@ impl ChainspecLoader {
             chainspec,
             root_dir,
             reactor_exit,
-            state_root_hash: None,
+            starting_state_root_hash: Digest::default(),
```

Comment:
Why would we ever reuse cached/stored state? Shouldn't we always use the one provided on the node startup?

Also, the `Digest::default()` is supposed to be a placeholder that will be replaced ones we handle the `Initialize` event?

Why not make this constructor `async` and do that in the `initializer.rs` constructor, even before we have a reactor.

Reply:
> Why would we ever reuse cached/stored state? Shouldn't we always use the one provided on the node startup?

This would be for restarting where we didn't stop for an upgrade.

> Also, the Digest::default() is supposed to be a placeholder that will be replaced ones we handle the Initialize event?

Yes.

> Why not make this constructor async and do that in the initializer.rs constructor, even before we have a reactor.

I don't think there's a need.  We don't have any other components' constructors async.  We have a design pattern of keeping the components methods synchronous and I'd rather keep to that.

Reply:
> I don't think there's a need. We don't have any other components' constructors async. We have a design pattern of keeping the components methods synchronous and I'd rather keep to that.

Sure, but the reactor's constructor is already async and I think it's worth making the component's constructor async to simplify the setup. Those `Initialize` events introduce a lot of (unnecessary) complexity. Just FYI, I will be using async constructors if they will simplify the code :)